fms/modules/attention.py

```diff
@@ -132,35 +133,35 @@ def forward(
             in past_key_value_state
         """
 
+        if k is None and v is None:
```

Comment:
I think you also want to handle `or (k is q and v is q)`. It's pretty typical for users to call these kinds of models like `attn(x, x, x)`.

Reply:
Yes, that would be good to check as well, I will add it.